src/sql/base/browser/ui/infoButton/infoButton.ts

```diff
@@ -0,0 +1,172 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import 'vs/css!./infoButton';
+import { Button as sqlButton } from 'sql/base/browser/ui/button/button';
+import * as DOM from 'vs/base/browser/dom';
+import { IButtonOptions } from 'vs/base/browser/ui/button/button';
+
+
+// type IUserFriendlyIcon = string | URI | { light: string | URI; dark: string | URI };
+
+export interface IInfoButtonOptions extends IButtonOptions {
+	buttonMaxHeight: number,
+	buttonMaxWidth: number,
+	description: string,
+	iconClass: string,
+	iconHeight: number,
+	iconWidth: number,
+	title: string,
+}
+
+export class InfoButton extends sqlButton {
+	private _container: HTMLElement;
+	private _main: HTMLElement;
+	private _iconContainer: HTMLElement;
+	private _iconElement: HTMLElement;
+	private _textContainer: HTMLElement;
+	private _pTitle: HTMLElement;
+	private _pDesc: HTMLElement;
+
+	private _buttonMaxHeight?: number;
+	private _buttonMaxWidth?: number;
+	private _description?: string;
+	private _iconClass?: string;
+	private _iconHeight?: number;
+	private _iconWidth?: number;
+	private _title?: string;
+
+	constructor(container: HTMLElement, options?: IInfoButtonOptions) {
+		super(container, options);
+		this._container = container;
+
+		{ // Creates the elements
```

Comment:
is there a reason you are putting the code here in a block?

Reply:
Just kept is separate to avoid confusion. Want me to remove that block format?

Reply:
yeah, we don't usually do that. lol

Reply:
Corrected.